### PR TITLE
use pip to install packages from requirements.txt

### LINKED ISSUE #46: requirements.txt parser generality
- [ ] Look over the [`requirements.txt` spec](https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format) and flag unsupported stuff
- [x] Support comments

cc @matthiasdiener

### LINKED ISSUE #58: Hack requirements.txt parsing to allow not producing a checkout of a package
cf. https://github.com/illinois-ceesd/mirgecom/pull/51

### LINKED ISSUE #108: Change requirements.txt parsing to use pip
Imho, at least for the medium-term, we should probably abandon the requirements.txt parsing completely, and just read the file line-by-line, feeding each line to pip directly. With a special case for the development packages such as loopy, where we just add a `--src` argument to pip.

_Originally posted by @matthiasdiener in https://github.com/illinois-ceesd/mirgecom/pull/244#issuecomment-776300112_

Fixes #46.

### LINKED ISSUE #111: Use mirgecom's conda env file for conda packages
instead of the hardcoded list in install-conda-dependencies

### PR DESCRIPTION
**Pull request by matthiasdiener**
instead of parsing requirements.txt ourselves

Closes #58, closes #46, fixes #108.

Requires https://github.com/illinois-ceesd/mirgecom/pull/252

### CODE REVIEW

File: .github/workflows/makezip.yaml [hunk 1]
```diff
@@ -13,9 +13,9 @@ jobs:
     - name: Build modules.zip
       run: |
         [[ $(uname) == Linux ]] && sudo apt-get install -y openmpi-bin libopenmpi-dev
-        ./install.sh
+        ./install.sh --branch=pip-editable
```
Comment (matthiasdiener) [inline .github/workflows/makezip.yaml:16]: These changes need to be undone before the final merge.

**Review by MTCam — DISMISSED** [review]
I think that without live-testing this, it's going to be hard to ferret out and/or avoid some fallout.   We've been talking about this for some time, and it reduces a good deal of opaque behavior in `emirge`.

File: version.sh [hunk 2]
```diff
@@ -135,27 +127,40 @@ echo "# Python: $(which python) [$(python --version)]"
...
 seen_mirgecom=0
 
-for i in "${!module_names[@]}"; do
-    url=${module_full_urls[$i]}
-    name=${module_names[$i]}
-    branch=${module_branches[$i]/--branch /}
-    giturl=${url/\#egg=[a-z]*/}
-    [[ ${url} =~ (#egg=[a-z]*) ]] && egg=${BASH_REMATCH[1]} || egg=""
-    [[ -z $url ]] && continue # Ignore non-Git modules
+for name in */; do
```
Comment (inducer) [inline version.sh:130]: Package enumeration is now worse (here and elsewhere): There's no authoritative source of information on what the packages managed by emirge actually are.

I think that's a marked downside of this direction.

File: version.sh [hunk 3]
```diff
@@ -135,27 +127,40 @@ echo "# Python: $(which python) [$(python --version)]"
...
+        cd "loo-py" || exit 1
+    else
+        cd "$name" || exit 1
+    fi
+
+    giturl=$(git config --get remote.origin.url)
+    if [[ $giturl == https://* ]]; then
+        giturl="git+$giturl"
+    else
+        giturl="git+ssh://$giturl"
```
Comment (inducer) [inline version.sh:158]: This isn't true AFAIK. `git+ssh://git@github.com:illinois-ceesd/mirgecom.git` is not a valid git URL.

File: version.sh (hunk 2 above)
Comment (matthiasdiener) [reply version.sh:130]: True, but the info was (kind of) inaccurate before as well - switching (e.g.) between mirgecom branches might change requirements.txt, but the packages installed/managed by emirge will not

File: version.sh (hunk 3 above)
Comment (matthiasdiener) [reply version.sh]: Good catch, fixed.

File: version.sh (hunk 2 above)
Comment (inducer) [reply version.sh:130]: :shrug: I can live with this. I don't love it, but I don't see a better option either. The parsing code was bad, too.

File: .github/workflows/makezip.yaml (hunk 1 above)
Comment (matthiasdiener) [reply .github/workflows/makezip.yaml]: Done.

**Review by MTCam — APPROVED** [review]
👍

**Pull request merged**